docs/overview.mdx

```diff
@@ -37,6 +37,64 @@ Casdoor serves both the web UI and the login requests from the application users
+
++ For `client_id`: you can find this under each single `Application`  
+
++ For `redirect_uri`: you should set this to your own `Application`'s callback URL, with this info, `Casdoor` can know where to send info back after authorization  
+
++ For `state`: you should fill this out with your `Application` name  
+
+The `Application` will speak to the user: *"Hey, now I need some resources and I need your permission to take these resources, you mind go to this `URL` and fill out your username and password for me?"*  
+
+With the correctly composed `URL`, your `Application` will make a user launch a request to this `URL`, and the `Authorization Request` is done.  
+
+### Step 2 (Authorization Grant) 
+
+This step is straightforward: the user is redirected to the `URL` composed above, and the user should see the login page from `Casdoor`.  
+By typing the correct username and credential into the login form, `Casdoor` now knows the identity of the user, and is about to send two keywords: `code` and `state` back to the callback URL set in Step 1.  
+
+The user opens the `URL` and provides the credentials to `Casdoor`. `Casdoor` will say: *"Looking good ~ this is the user (who is authorizing the `Application` to get the `code` and `state`) I know in my database, and I will send the `code` and `state` back to the `Application` using the callback URL (`redirect_uri`)"*  
+
+With these two keywords send back to your `Application`, the authrization is now granted to the app, and thus `Authorization Grant` is completed.  
+
+### Step 3 (Authorization Grant) 
+
+In this step, your `Application` already has the `code` in hand from Step 2, and it will speak to `Casdoor`: *"Hey, now the user agreed to give me the `code`, you wanna check this `code` out and give me the `access_token`?"*  
+
+### Step 4 (Access Token) 
+
+In this step, `Casdoor` speaks back to the `Application`: *"You know what, this `code` seems legit, you must be the right `Application`. Come, here's the `access_token`."*  
+With this `code`, that `Casdoor` knows that it is an authorized `Application` (authorization given by the correct user in Step 2) trying to get the `access_token` (will be used later to get more useful things).
+
+### Step 5 (Access Token)
+
+In this step, your `Application` says: *"Nice one, just got the fresh-and-tasty `access_token`, I can now use it to get something more valuable from the `Resource Server`!"*  
+
+And your `Application` turns to the `Resource Server`: *"Hey buddy, wanna check this `access_token` out? I got this from `Casdoor` and you want to see if this is the correct one you had with `Casdoor`?"*  
+
+### Step 6 (Protected Resource)
+
+The `Resource Server` speaks back to your `Application`: *"Not bad ~ it seems just like the one I had with `Casdoor`, and `Casdoor` says whoever hold this `access_token` can have these `Protected Resource`. Now, you take it!"*  
+
+And this is basically how `Casdoor` works with your `Application`.
+
+**Hint:**
+
```

Comment:
Same as the usage of admonitions.

Reply:
will do

Reply:
> Please don’t use too much code emphasis, bold face letter is also a good choice.

Here is a list of keywords that I keep the code format:
`code`
`access_token`
`endpoint`
`client_id`
`redirect_uri`

and the OAuth 2.0 procedure names.